src/evaluation/stats.jl

```diff
@@ -170,3 +172,70 @@ function aggregate_stats(df::DataFrame, criterion_col=:val_auc;
 	end
 	vcat(results...)
 end
+
+
+"""
+aggregate_stats_max_mean(df::DataFrame, criterion_col=:val_auc; undersample=Dict("ocsvm" => 100))
+
+Chooses the best hyperparameters for each seed/anomaly_class combination by `criterion_col`
+and then aggregates the metrics over seed/anomaly_class to get the final results. The output 
+is a DataFrame of maximum #datasets*#models rows with
+columns of different types
+- identifiers - `dataset`, `modelname`, `phash`, `parameters`
+- averaged metrics - both from test and validation data such as `tst_auc`, `val_pat_10`, etc.
+- std of best hyperparameter computed for each metric over different seeds, suffixed `_std`
+- std of best 10 hyperparameters computed over averaged metrics, suffixed `_top_10_std`
+- samples involved in the aggregation, 
++ `psamples` - number of runs of the best hyperparameter
++ `dsamples` - number of sampled hyperparameters
++ `dsamples_valid` - number of sampled hyperparameters with enough runs
+When nonempty `downsample` dictionary is specified, the entries of`("model" => #samples)`, specify
+how many samples should be taken into acount. These are selected randomly with fixed seed.
+TODO: Optional arg `min_samples` specifies how many seed/anomaly_class combinations should be present
+in order for the hyperparameter's results be considered statistically significant.
+"""
+function aggregate_stats_max_mean(df::DataFrame, criterion_col=:val_auc; 
+                        min_samples=("anomaly_class" in names(df)) ? 30 : 3, 
+                        downsample=Dict("ocsvm" => 100))
+agg_cols = vcat(_prefix_symbol.("val", BASE_METRICS), _prefix_symbol.("tst", BASE_METRICS))
+agg_cols = vcat(agg_cols, _prefix_symbol.("val", PAT_METRICS), _prefix_symbol.("tst", PAT_METRICS))
+top10_std_cols = _prefix_symbol.(agg_cols, "top_10_std")
+
+agg_keys = ("anomaly_class" in names(df)) ? [:seed, :anomaly_class] : [:seed]
+# choose best for each seed/anomaly_class cobination and average over them
+results = []
+for (dkey, dg) in pairs(groupby(df, :dataset))
+    for (mkey, mg) in pairs(groupby(dg, :modelname))
+        partial_results = []
```

Comment:
If we have agreed on seed 1 only, you should filter here somehow.

Reply:
I hesitate to put such filter into the aggregation itself and leave this to the user to filter beforehand.